library/core/registry/src/main/java/org/quiltmc/qsl/registry/impl/sync/server/ServerRegistrySync.java

```diff
@@ -0,0 +1,326 @@
+import org.quiltmc.qsl.networking.api.PacketByteBufs;
+import org.quiltmc.qsl.networking.api.ServerPlayNetworking;
+import org.quiltmc.qsl.registry.api.sync.RegistrySynchronization;
+import org.quiltmc.qsl.registry.impl.RegistryConfig;
+import org.quiltmc.qsl.registry.impl.sync.ProtocolVersions;
+import org.quiltmc.qsl.registry.impl.sync.ServerPackets;
+import org.quiltmc.qsl.registry.impl.sync.modprotocol.ModProtocolDef;
+import org.quiltmc.qsl.registry.impl.sync.modprotocol.ModProtocolImpl;
+import org.quiltmc.qsl.registry.impl.sync.registry.SynchronizedRegistry;
+import org.quiltmc.qsl.registry.impl.sync.registry.RegistryFlag;
+
+@ApiStatus.Internal
+public final class ServerRegistrySync {
+	private static final int MAX_SAFE_PACKET_SIZE = 734003;
+
+	public static Text noRegistrySyncMessage = Text.empty();
+	public static Text errorStyleHeader = Text.empty();
+	public static Text errorStyleFooter = Text.empty();
+	public static boolean supportFabric = false;
+	public static boolean forceFabricFallback = false;
+	public static boolean forceDisable = false;
+	public static boolean showErrorDetails = false;
+
+	public static IntList SERVER_SUPPORTED_PROTOCOL = new IntArrayList(ProtocolVersions.IMPL_SUPPORTED_VERSIONS);
+
+	public static void readConfig() {
+		//var config = RegistryConfig.INSTANCE.registry_sync;
+
+		//noRegistrySyncMessage = text(config.missing_registry_sync_message);
+		noRegistrySyncMessage = text((String) RegistryConfig.getSync("missing_registry_sync_message"));
+		//errorStyleHeader = text(config.mismatched_entries_top_message);
+		errorStyleHeader = text((String) RegistryConfig.getSync("mismatched_entries_top_message"));
+		//errorStyleFooter = text(config.mismatched_entries_bottom_message);
+		errorStyleFooter = text((String) RegistryConfig.getSync("mismatched_entries_bottom_message"));
+
+		//supportFabric = config.support_fabric_api_protocol;
+		supportFabric = (Boolean) RegistryConfig.getSync("support_fabric_api_protocol");
+		//forceFabricFallback = config.force_fabric_api_protocol_fallback;
+		forceFabricFallback = (Boolean) RegistryConfig.getSync("force_fabric_api_protocol_fallback");
+		//forceDisable = config.disable_registry_sync;
+		forceDisable = (Boolean) RegistryConfig.getSync("disable_registry_sync");
+		//showErrorDetails = config.mismatched_entries_show_details;
+		showErrorDetails = (Boolean) RegistryConfig.getSync("mismatched_entries_show_details");
+
+		if (supportFabric) {
+			SERVER_SUPPORTED_PROTOCOL.add(ProtocolVersions.FAPI_PROTOCOL);
+		}
+
+		if (forceDisable) {
+			SERVER_SUPPORTED_PROTOCOL.clear();
+		}
+	}
+
+	public static void requireMinimumVersion(int version) {
+		SERVER_SUPPORTED_PROTOCOL.removeIf(x -> x < version);
+	}
+
+	private static Text text(String string) {
+		if (string == null) {
+			return Text.empty();
+		}
+
+		Text text = null;
+		try {
+			text = Text.Serializer.fromJson(string);
+		} catch (Exception e) {
+		}
+
+		if (text == null) {
+			text = Text.literal(string);
+		}
```

Comment:
Actually, maybe this could be done?
```suggestion
		var text = text = Text.literal(string);

		try {
			text = Text.Serializer.fromJson(string);
		} catch (Exception e) {}
```

Reply:
This won't do, as technically fromJson might return null without throwing

Reply:
ah, that makes sense